lib/alice/adapters.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Alice.Adapters do
+  @moduledoc false
+
+  def start_link(opts \\ [start_adapter: true]) do
+    if opts[:start_adapter], do: selected_adapter.start_link
+    {:ok, selected_adapter}
+  end
+
+  @doc """
+  Selects adapter, defaults to Slack
+  """
+  def selected_adapter do
+    Application.get_env(:alice, :adapter, Alice.Adapters.Slack)
```

Comment:
@tapickell actually now that I think of it, there may be some issues doing that. A few processes are registering global names. This may be something that we need to look into supporting. Shouldn't be too difficult, though. Each Alice instance would just need to be given a unique name.